tests/br_db/run.sh

```diff
@@ -36,7 +39,20 @@ run_sql "CREATE TABLE $DB.usertable2 ( \
 run_sql "INSERT INTO $DB.usertable2 VALUES (\"c\", \"d\");"
 # backup db
 echo "backup start..."
+export GO_FAILPOINTS="github.com/pingcap/br/pkg/task/progress-call-back=return(\"$PROGRESS_FILE\")"
 run_br --pd $PD_ADDR backup db --db "$DB" -s "local://$TEST_DIR/$DB" --ratelimit 5 --concurrency 4
+export GO_FAILPOINTS=""
+
+# check if we use the region unit
+if [[ "$(wc -l <$PROGRESS_FILE)" == "1" ]] && [[ !$(grep -q "region" $PROGRESS_FILE) ]];
```

Comment:
I'm afraid `[[ !$(grep -q "region" $PROGRESS_FILE) ]]` would be `true` in any condition:

Firstly, the expression would be evaluated to: `[[ ! ]]`, since `grep -q` wouldn't print anything to stdout. And its exit code would be ignored in this context.

Then, with single argument, `[[` would test whether the argument is empty string(i.e. `[[ -n ! ]]`), and '!' isn't empty, so the result of test would always be `true`.

Reply:
Ok, fixed.

Now it can be tested. https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/br_ghpr_unit_and_integration_test/detail/br_ghpr_unit_and_integration_test/10502/pipeline